### PR TITLE
Fix/ink 4 0 beta test

### PR DESCRIPTION
**Pull request by HyunggyuJang**
- Fixed all the tests
- Update typechain-polkadot
- Lint configuration
- Update test infrastructure

### CODE REVIEW

File: uniswap-v2/contracts/psp22/lib.rs [hunk 1]
```diff
@@ -71,15 +71,24 @@ pub mod token {
...
+            instance.metadata.name = name;
+            instance.metadata.symbol = symbol;
+            instance.metadata.decimals = decimals;
             instance
                 ._mint_to(instance.env().caller(), total_supply)
                 .expect("Should mint");
             instance
         }
+        #[ink(message)]
+        pub fn mint(&mut self, account: AccountId, amount: Balance) -> Result<(), PSP22Error> {
```
Comment (PierreOssun) [inline uniswap-v2/contracts/psp22/lib.rs:92]: Why is this needed ?

Comment (HyunggyuJang) [reply uniswap-v2/contracts/psp22/lib.rs:92]: Ah, mint actually isn't used in the integration test. Permissionless mint, however, is necessary to let user test our dex template deployed on Shibuya.

Comment (HyunggyuJang) [reply uniswap-v2/contracts/psp22/lib.rs:92]: Similar reason to https://github.com/AstarNetwork/wasm-showcase-dapps/pull/34#discussion_r1040711707

Comment (PierreOssun) [reply uniswap-v2/contracts/psp22/lib.rs:92]: Then can you add a comment that this should only be used for test purpose please

Comment (HyunggyuJang) [reply uniswap-v2/contracts/psp22/lib.rs:92]: Got it!

Comment (HyunggyuJang) [reply uniswap-v2/contracts/psp22/lib.rs:92]: Done

**Review by PierreOssun — APPROVED** [review]
LGTM
